### PR TITLE
Add support for custom task runner

### LINKED ISSUE #8067: Allow Custom Task Runner via plugins
**Description**
Airflow provides a plugin manager to integrate external features. However, there is no configuration available to have custom task runner as part of plugins.

**Use case / motivation**
 Enabling custom task runner integration with Airflow will help users in configuring runners as per environment requirement and dependencies.

### PR DESCRIPTION
**Pull request by mik-laj**
https://github.com/apache/airflow/issues/8067
 I'm not sure if anyone will use it, but it can stimulate community creativity.

---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/task/task_runner/__init__.py [hunk 1]
```diff
@@ -35,10 +46,12 @@ def get_task_runner(local_task_job):
...
-        from airflow.task.task_runner.cgroup_task_runner import CgroupTaskRunner
-        return CgroupTaskRunner(local_task_job)
-    else:
-        raise AirflowException("Unknown task runner type {}".format(_TASK_RUNNER))
+    if _TASK_RUNNER_NAME in CORE_TASK_RUNNERS:
+        log.debug("Loading core task runner: %s", _TASK_RUNNER_NAME)
+        task_runner_class = import_string(CORE_TASK_RUNNERS[_TASK_RUNNER_NAME])
+        return task_runner_class(local_task_job)
+
+    log.debug("Loading task runner from custom path: %s", _TASK_RUNNER_NAME)
```
Comment (rabsr) [inline airflow/task/task_runner/__init__.py]: Please remove code duplication

**Review by rabsr — CHANGES_REQUESTED** [review]
Also, import correction for StandardTaskRunner in test_standard_task_runner is required

File: airflow/task/task_runner/__init__.py (hunk 1 above)
Comment (mik-laj) [reply airflow/task/task_runner/__init__.py]: We don't have code duplication. These are separate ways to load modules. 

There is only one common fragment.
``
task_runner_class(local_task_job)
``
However, pulling out this common part will not reduce the amount of code.

Comment (rabsr) [reply airflow/task/task_runner/__init__.py]: The only difference I see is that the parameter pass to `import_string`. Rest code can be taken out in a method passing the string parameter required by `import_string` to new method.

Comment (rabsr) [reply airflow/task/task_runner/__init__.py]: Something like this 
```
def get_runner(runner_class, local_task_job):
    log.debug("Loading core task runner: %s", _TASK_RUNNER_NAME)
    task_runner_class = import_string(runner_class)
    return task_runner_class(local_task_job)


def get_task_runner(local_task_job):
    if _TASK_RUNNER_NAME in CORE_TASK_RUNNERS:
        return get_runner(CORE_TASK_RUNNERS[_TASK_RUNNER_NAME], local_task_job)
    return get_runner(_TASK_RUNNER_NAME, local_task_job)

```

**Review by rabsr — APPROVED** [review]

**Review by houqp — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
LGTM. Do you have already some other task runner you think about?

**Comment by mik-laj** [comment]
@potiuk Yes. Databand wants to influence the way tasks are started. One user has also reported such a need.
https://github.com/apache/airflow/issues/8067

**Pull request merged**